students/Deniss_Semcovs/Lesson04/mailroom_part2.py

```diff
@@ -26,52 +52,109 @@ def sendTYnote():
     print(("-")*61)
     for i in data:
-        dName = (i[0])
-        tGiven = sum(i[1:])
-        nGifts = len(i[1:])
-        aGift = ((sum(i[1:]))/(len(i[1:])))
-        data2 = (dName, tGiven, nGifts, aGift)
+        dname = (i[0])
+        tgiven = sum(i[1:])
+        ngifts = len(i[1:])
+        agift = ((sum(i[1:]))/(len(i[1:])))
+        data2 = (dname, tgiven, ngifts, agift)
         print("{:20s} $ {:10d}   {:10d}  $ {:.2f}".format(*data2))
 
-def sendTYnoteALL():
+def send_ty_note_all():
     for i in data:
-	    name = i[0]
+	    filename = (i[0]+".txt")
 		tDonation = sum(i[1:])
-		print(" ")
-		print("Hello {}, thank you for your generous donation of ${} to support our cause.".format(name, tDonation))
+		import tempfile
+		import os.path
+		location = tempfile.gettempdir()
+		lfile = os.path.join(location, filename)
+		with open(lfile, "w") as f:
+		    f.write("\nDear "+i[0]+",")
+			f.write("\n"*2)
+			f.write(" "*8+"Thank you for your very kind donation of $"+str(tDonation)+".")
+			f.write("\n"*2)
+			f.write(" "*8+"It will be put to very good use.")
+			f.write("\n"*2)
+			f.write(30*" "+"Sincerely,")
+			f.write(34*" "+"-The Team")
 
 def quit():
     print("Quiting this menu now!")
 	return "exit menu"
 
+def send_ty_note():
+    mailroom02(send_ty_note_reply,send_ty_note_action)
+
+def don01_letter():
+	print(letter.format(**don01))
+
+
+def don02_letter():
+	print(letter.format(**don02))
+
+def don03_letter():
+	print(letter.format(**don03))
+
+def don04_letter():
+	print(letter.format(**don04))
+
+def don05_letter():
+	print(letter.format(**don05))
+
+def don06_letter():
+	print(letter.format(**don06))
+
+letter = ("Hello {name}, thank you for your generous donation of ${don} to support our cause.")
 main_reply = ("\nChoose an action:\n"
                "\n"
-			   "1 - Send a Thank You to a single donor.\n"
-			   "2 - Create a Report.\n"
-			   "3 - Send letters to all donors.\n"
-			   "4 - Quit\n"
+			   "1 - Send a Thank You note to a single donor.\n"
+			   "2 - Add new donation.\n"
+			   "3 - Create a Report.\n"
+			   "4 - Send letters to all donors.\n"
+			   "5 - To see the LIST of donor names.\n"
+			   "q - Quit\n"			   
 			   ">>>")
 
-main_action = {"1":sendTYnote,
-               "2":createReport,
-			   "3":sendTYnoteALL,
-			   "4":quit
+main_action = {"1":send_ty_note,
+               "2":add_donation,
+               "3":create_report,
+			   "4":send_ty_note_all,
+			   "5":print_list,
+			   "q":quit
 			   }
+send_ty_note_reply = ("\nPlease enter full name.\n"
+                    "q - to exit this menu.\n" 
+                    ">>>")
+send_ty_note_action = {"Tom Hanks":don01_letter,
+                     'Leonardo DiCaprio':don02_letter,
+					 'Natalie Portman':don03_letter,
+					 'Harrison Ford':don04_letter,
+					 'Scarlett Johansson':don05_letter,
+					 'Jennifer Aniston':don06_letter,
+					 "q":quit
+                     }
+argument = 1
+don01={'name':'Tom Hanks','don':1500}
```

Comment:
There should be whitespaces around assignment operator as well as colon - for here and all below.